contrib/pg_tde/src/catalog/tde_principal_key.c

```diff
@@ -598,12 +615,52 @@ pg_tde_set_principal_key_internal(char *principal_key_name, int is_global, char
 		providerOid = GLOBAL_DATA_TDE_OID;
 		dbOid = GLOBAL_DATA_TDE_OID;
 	}
+
+	if (is_global == 3)			/* using a globla provider for the default
```

Comment:
Make sense to name these constants (1,2,3) and many rename is_global to global_type or something like that. So it's easier to read the code.

Reply:
I suggest creating a C enum for this.

Reply:
Done.